packages/circuits/src/BaseProof.ts

```diff
@@ -13,8 +12,8 @@ export class BaseProof {
     readonly _snarkProof: SnarkProof
     protected circuit?: Circuit
 
-    readonly publicSignals: bigint[]
-    public proof: bigint[]
+    readonly publicSignals: string[]
+    public proof: string[]
```

Comment:
Why are we changing this type from `bigint[]` to `string[]`?

Reply:
now when we developing attester apps (frontend)
it requires to change types to send proofs to backend
I think `string` is more convenient and it is originally `string` when proof is generated

https://github.com/Unirep/create-unirep-app/blob/833685b2b8b15a63ddbe114b79f2a333e3f0bf43/packages/frontend/src/contexts/User.ts#L160-L165

Reply:
I tend to use `bigint` values everywhere when i work with this library. I think `string` values are ambiguous, they can be e.g. decimal or hex or binary. It's kinda okay because the `string` must be a valid input to `BigInt(str)`, but this is only checked when/if `bigint` conversion happens.

I also think there may be a cost to parsing the `string` into a `bigint`, e.g. serialization.

Reply:
[1658f01](https://github.com/Unirep/Unirep/pull/547/commits/1658f01f89d47cf25d29314f24ce83aa24a5fddf)